scipy/signal/_spectral_py.py

```diff
@@ -1976,16 +2010,40 @@ def _fft_helper(x, win, detrend_func, nperseg, noverlap, nfft, sides):
 
     .. versionadded:: 0.16.0
     """
+    xp = array_namespace(x)
+    x = xp.asarray(x)
     # Created strided array of data segments
     if nperseg == 1 and noverlap == 0:
-        result = x[..., np.newaxis]
+        result = x[..., xp.newaxis]
     else:
         # https://stackoverflow.com/a/5568169
         step = nperseg - noverlap
         shape = x.shape[:-1]+((x.shape[-1]-noverlap)//step, nperseg)
-        strides = x.strides[:-1]+(step*x.strides[-1], x.strides[-1])
-        result = np.lib.stride_tricks.as_strided(x, shape=shape,
-                                                 strides=strides)
+        if hasattr(xp, "device"):
+            result = xp.empty(shape, dtype=x.dtype, device=xp.device(x))
+        else:
+            result = xp.empty(shape, dtype=x.dtype)
+        if "torch" in xp.__name__:
+            strides = x.stride()[:-1]+(step*x.stride()[-1], x.stride()[-1])
+            result = xp.as_strided(x, size=shape, stride=strides)
+        elif "cupy" in xp.__name__:
+            strides = x.strides[:-1]+(step*x.strides[-1], x.strides[-1])
+            result = xp.lib.stride_tricks.as_strided(x, shape=shape,
```

Comment:
As mentioned above, I think the special-casing should be saved for a separate PR.